lldb/tools/lldb-dap/ProgressEvent.cpp

```diff
@@ -164,10 +168,11 @@ const ProgressEvent &ProgressEventManager::GetMostRecentEvent() const {
   return m_last_update_event ? *m_last_update_event : m_start_event;
 }
 
-void ProgressEventManager::Update(uint64_t progress_id, uint64_t completed,
-                                  uint64_t total) {
-  if (std::optional<ProgressEvent> event = ProgressEvent::Create(
-          progress_id, std::nullopt, completed, total, &GetMostRecentEvent())) {
+void ProgressEventManager::Update(uint64_t progress_id, const char *message,
+                                  uint64_t completed, uint64_t total) {
+  if (std::optional<ProgressEvent> event =
+          ProgressEvent::Create(progress_id, StringRef(message), completed,
```

Comment:
```suggestion
          ProgressEvent::Create(progress_id, message, completed,
```

Reply:
The constructor actually takes an optional of a string ref. It felt clunkier to me to pass down the optional through the update call than just call stringref when creating the object in the update call. I think a null string ref should be fine, but I'll validate.